From ab7a01df87b50de4d9cd0c40ffae43ded219bad5 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Fri, 30 Mar 2018 13:47:54 +0300
Subject: [PATCH 431/767] sdk: dts: ppc: include CEETM nodes separately based
 on SoC type

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../dts/fsl/qoriq-qman3-ceetm0-32-sdk.dtsi    | 42 +++++++++++++++++++
 .../dts/fsl/qoriq-qman3-ceetm0-8-sdk.dtsi     | 42 +++++++++++++++++++
 .../dts/fsl/qoriq-qman3-ceetm1-32-sdk.dtsi    | 42 +++++++++++++++++++
 .../boot/dts/fsl/qoriq-qman3-portals-sdk.dtsi |  7 ----
 arch/powerpc/boot/dts/fsl/t1023rdb-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t1024qds-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t1024rdb-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t1040d4rdb-sdk.dts  |  1 +
 arch/powerpc/boot/dts/fsl/t1040qds-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t1040rdb-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t1042d4rdb-sdk.dts  |  1 +
 arch/powerpc/boot/dts/fsl/t1042rdb-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t2080qds-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t2080rdb-sdk.dts    |  1 +
 arch/powerpc/boot/dts/fsl/t4240qds-sdk.dts    |  1 +
 15 files changed, 137 insertions(+), 7 deletions(-)
 create mode 100644 arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-32-sdk.dtsi
 create mode 100644 arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-8-sdk.dtsi
 create mode 100644 arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm1-32-sdk.dtsi

diff --git a/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-32-sdk.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-32-sdk.dtsi
new file mode 100644
index 000000000000..02474c9e1601
--- /dev/null
+++ b/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-32-sdk.dtsi
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2012 - 2015 Freescale Semiconductor Inc.
+ * Copyright 2017 - 2018 NXP
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+&qportals {
+	qman-ceetm@0 {
+		compatible = "fsl,qman-ceetm";
+		fsl,ceetm-lfqid-range = <0xf00000 0x1000>;
+		fsl,ceetm-sp-range = <0 12>;
+		fsl,ceetm-lni-range = <0 8>;
+		fsl,ceetm-channel-range = <0 32>;
+	};
+};
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-8-sdk.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-8-sdk.dtsi
new file mode 100644
index 000000000000..66cb8d94decd
--- /dev/null
+++ b/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm0-8-sdk.dtsi
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2012 - 2015 Freescale Semiconductor Inc.
+ * Copyright 2017 - 2018 NXP
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+&qportals {
+	qman-ceetm@0 {
+		compatible = "fsl,qman-ceetm";
+		fsl,ceetm-lfqid-range = <0xf00000 0x1000>;
+		fsl,ceetm-sp-range = <0 12>;
+		fsl,ceetm-lni-range = <0 8>;
+		fsl,ceetm-channel-range = <0 8>;
+	};
+};
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm1-32-sdk.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm1-32-sdk.dtsi
new file mode 100644
index 000000000000..91bb5f847710
--- /dev/null
+++ b/arch/powerpc/boot/dts/fsl/qoriq-qman3-ceetm1-32-sdk.dtsi
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2012 - 2015 Freescale Semiconductor Inc.
+ * Copyright 2017 - 2018 NXP
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+&qportals {
+	qman-ceetm@1 {
+		compatible = "fsl,qman-ceetm";
+		fsl,ceetm-lfqid-range = <0xf10000 0x1000>;
+		fsl,ceetm-sp-range = <0 12>;
+		fsl,ceetm-lni-range = <0 8>;
+		fsl,ceetm-channel-range = <0 32>;
+	};
+};
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-qman3-portals-sdk.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-qman3-portals-sdk.dtsi
index 64ad22d27ba5..d8cb5390a305 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-qman3-portals-sdk.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-qman3-portals-sdk.dtsi
@@ -48,11 +48,4 @@
 		compatible = "fsl,cgrid-range";
 		fsl,cgrid-range = <0 256>;
 	};
-	qman-ceetm@0 {
-		compatible = "fsl,qman-ceetm";
-		fsl,ceetm-lfqid-range = <0xf00000 0x1000>;
-		fsl,ceetm-sp-range = <0 12>;
-		fsl,ceetm-lni-range = <0 8>;
-		fsl,ceetm-channel-range = <0 32>;
-	};
 };
diff --git a/arch/powerpc/boot/dts/fsl/t1023rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t1023rdb-sdk.dts
index 70ed1b9ed1a7..0f47d629b826 100644
--- a/arch/powerpc/boot/dts/fsl/t1023rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1023rdb-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t1024qds-sdk.dts b/arch/powerpc/boot/dts/fsl/t1024qds-sdk.dts
index 599f5321308c..b79b03fda3fb 100644
--- a/arch/powerpc/boot/dts/fsl/t1024qds-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1024qds-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t1024rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t1024rdb-sdk.dts
index 7fae26fc82c6..aba160e00cd0 100644
--- a/arch/powerpc/boot/dts/fsl/t1024rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1024rdb-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t1040d4rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t1040d4rdb-sdk.dts
index 72279eff5a6d..d640741b1eba 100644
--- a/arch/powerpc/boot/dts/fsl/t1040d4rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1040d4rdb-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t1040qds-sdk.dts b/arch/powerpc/boot/dts/fsl/t1040qds-sdk.dts
index 794606baa555..0cb52f84df38 100644
--- a/arch/powerpc/boot/dts/fsl/t1040qds-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1040qds-sdk.dts
@@ -3,3 +3,4 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/t1040rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t1040rdb-sdk.dts
index 002177c40e6b..41a6f0a244ed 100644
--- a/arch/powerpc/boot/dts/fsl/t1040rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1040rdb-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t1042d4rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t1042d4rdb-sdk.dts
index 15d5f8baf60b..31f9143635c3 100644
--- a/arch/powerpc/boot/dts/fsl/t1042d4rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1042d4rdb-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t1042rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t1042rdb-sdk.dts
index f5e9c67eb8ca..c8c00b9cecc6 100644
--- a/arch/powerpc/boot/dts/fsl/t1042rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t1042rdb-sdk.dts
@@ -4,6 +4,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-8-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t2080qds-sdk.dts b/arch/powerpc/boot/dts/fsl/t2080qds-sdk.dts
index d03627ddfb50..e2e391964877 100644
--- a/arch/powerpc/boot/dts/fsl/t2080qds-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t2080qds-sdk.dts
@@ -4,3 +4,4 @@
 /include/ "qoriq-dpaa-eth.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-32-sdk.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/t2080rdb-sdk.dts b/arch/powerpc/boot/dts/fsl/t2080rdb-sdk.dts
index ddf835d1d2a0..c69f32b6896b 100644
--- a/arch/powerpc/boot/dts/fsl/t2080rdb-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t2080rdb-sdk.dts
@@ -3,6 +3,7 @@
 /include/ "qoriq-fman3-0-sdk.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-32-sdk.dtsi"
 
 &soc {
 	fsldpaa: fsl,dpaa {
diff --git a/arch/powerpc/boot/dts/fsl/t4240qds-sdk.dts b/arch/powerpc/boot/dts/fsl/t4240qds-sdk.dts
index 93ff3962672f..8e9860915124 100644
--- a/arch/powerpc/boot/dts/fsl/t4240qds-sdk.dts
+++ b/arch/powerpc/boot/dts/fsl/t4240qds-sdk.dts
@@ -4,3 +4,4 @@
 /include/ "qoriq-dpaa-eth.dtsi"
 /include/ "qoriq-bman-portals-sdk.dtsi"
 /include/ "qoriq-qman3-portals-sdk.dtsi"
+/include/ "qoriq-qman3-ceetm0-32-sdk.dtsi"
-- 
2.17.0

